instrumentation/nginx/src/otel_ngx_module.cpp

```diff
@@ -65,6 +66,62 @@ nostd::string_view NgxHttpFlavor(ngx_http_request_t* req) {
   }
 }
 
+static void NgxNormalizeAndCopyString(u_char* dst, ngx_str_t str) {
+  for (ngx_uint_t i = 0; i < str.len; ++i) {
+    u_char ch = str.data[i];
+    if (ch >= 'A' && ch <= 'Z') {
+      ch += 0x20;
+    } else if (ch == '-') {
+      ch = '_';
+    }
+
+    dst[i] = ch;
+  }
+
+  dst[str.len] = '\0';
+}
+
+static void OtelCaptureHeaders(nostd::shared_ptr<opentelemetry::trace::Span> span, ngx_str_t keyPrefix, ngx_list_t *headers,
+                        ngx_regex_t *sensitiveHeaderNames, ngx_regex_t *sensitiveHeaderValues,
+                        std::vector<ngx_table_elt_t*> excludedHeaders = {}) {
```

Comment:
Would passing `std::vector` passing by value here create an unnecessary heap alloc / free? Can't say without testing, but perhaps `std::array` would work here to avoid the potential allocation?

Reply:
Using std::array ist not possible, because the array size is part if the type (Maybe if I convert `OtelCaptureHeaders` to a template method). I used nosed::span now.

Reply:
Right, thanks :smile: